File: src/AddIns/BackendBindings/VBNetBinding/Project/Src/VBNetProjectBinding.cs
// Copyright (c) AlphaSierraPapa for the SharpDevelop Team (for details please see \doc\copyright.txt)
// This code is distributed under the GNU LGPL (for details please see \doc\license.txt)

using System;
using System.Xml;
using ICSharpCode.SharpDevelop.Internal.Templates;
using ICSharpCode.SharpDevelop.Project;

namespace ICSharpCode.VBNetBinding
{
	public class VBNetProjectBinding : IProjectBinding
	{
		public const string LanguageName = "VBNet";
		
		public string Language
		{
			get {
				return LanguageName;
			}
		}
		
		public IProject LoadProject(ProjectLoadInformation info)
		{
			return new VBNetProject(info);
		}
		
		public IProject CreateProject(ProjectCreateInformation info)
		{
			return new VBNetProject(info);
		}
		
		public bool HandlingMissingProject {
			get { 
				return false; 
			}
		}
	}
}

File: src/AddIns/BackendBindings/VBNetBinding/Project/VBNetProjectBinding.vb
' Copyright (c) AlphaSierraPapa for the SharpDevelop Team (for details please see \doc\copyright.txt)
' This code is distributed under the GNU LGPL (for details please see \doc\license.txt)

Imports System.Xml
Imports ICSharpCode.SharpDevelop.Project

Public Class VBNetProjectBinding
	Implements IProjectBinding
	Public Const LanguageName As String = "VBNet"

	Public ReadOnly Property Language() As String Implements IProjectBinding.Language
		Get
			Return LanguageName
		End Get
	End Property

	Public Function LoadProject(info As ProjectLoadInformation) As IProject Implements IProjectBinding.LoadProject
		Return New VBNetProject(info)
	End Function

	Public Function CreateProject(info As ProjectCreateInformation) As IProject Implements IProjectBinding.CreateProject
		Return New VBNetProject(info)
	End Function

	Public ReadOnly Property HandlingMissingProject() As Boolean Implements IProjectBinding.HandlingMissingProject
		Get
			Return False
		End Get
	End Property
End Class
